MapboxNavigation/NavigationMapView.swift

```diff
@@ -289,6 +295,11 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         if (tracksUserCourse) {
             updateCourseTracking(location: userLocationForCourseTracking, camera:self.camera, animated: false)
         }
+    
+        // TODO: Add style observation to be able to re-add highlighted building whenever it happens.
```

Comment:
Turns out removing styleObserver in `deinit` cases crash. Since for now functionality which allows to re-add buildings when style is changed is not ready yet I've removed observer completely and added `TODO` statement =.

Reply:
The map view doesn‘t take care of readding other things like the route line when the style changes, either, so I don’t think readding the buildings should be the map view’s responsibility. RouteMapViewController is currently responsible for readding the route line, so we should piggyback on that style observer to readd the buildings there.

Reply:
I'll move `FIXME` statement to `RouteMapViewController` to find out best way to re-add highlighted building layer there. Will create ticket for that as well.